core/src/main/java/io/grpc/util/RoundRobinLoadBalancerFactory.java

```diff
@@ -198,28 +211,68 @@ public void handleSubchannelState(Subchannel subchannel, ConnectivityStateInfo s
       if (stateInfo.getState() == IDLE) {
         subchannel.requestConnection();
       }
-      getSubchannelStateInfoRef(subchannel).value = stateInfo;
-      updateBalancingState(getAggregatedState(), getAggregatedError());
+      stateInfoRef.value = stateInfo;
+      updateBalancingState();
+    }
+
+    private void shutdownSubchannel(Subchannel subchannel) {
+      subchannel.shutdown();
+      getSubchannelStateInfoRef(subchannel).value =
+          ConnectivityStateInfo.forNonError(SHUTDOWN);
+      if (stickinessState != null) {
+        stickinessState.remove(subchannel);
```

Comment:
This branch is [not covered](https://coveralls.io/builds/18764189/source?filename=core%2Fsrc%2Fmain%2Fjava%2Fio%2Fgrpc%2Futil%2FRoundRobinLoadBalancerFactory.java#L223).

Reply:
I've extended a unit test to cover this case - name resolver removal of a sticky subchannel.